pyfoxsi/src/pyfoxsi/psf/psf.py

```diff
@@ -0,0 +1,187 @@
+        variables like `var1`.
+    amplitude : array_like
+        The amplitudes of the peaks.
+    xo : float
+        The center location in the x-axis.
+    yo : float
+        The center location in the y-axis.
+    sigma_x : array_like
+        The width in the unrotated x-direction.
+    sigma_y : array_like
+        The width in the unrotated y-direction.
+    theta : float (radian)
+        The rotation angle for each gaussian.
+
+    See Also
+    --------
+    gauss2d : relationship (optional)
+
+    Returns
+    -------
+    array
+        The amplitude of the function at each (x,y)
+
+    Examples
+    --------
+    These are written in doctest format, and should illustrate how to
+    use the function.
+    >>> x, y = np.meshgrid(np.arange(-10,10,1), np.arange(-10,10,1))
+    >>> data = multi_gauss2d((x, y), (100,10,1), 0, 0, (1,2,3), (5,6,7), np.pi/4.)
+    """
+    i = 0
+    for amp, sig_x, sig_y in zip(amplitude, sigma_x, sigma_y):
+        g = gauss2d((x, y), amp, center[0], center[1], sig_x, sig_y, theta)
+        if i == 0:
+            result = g
+        else:
+            result += g
+        i += 1
+    return result
+
+def psf(x, y, scale=1 * u.arcsec / u.pix):
+    r"""The point spread function.
+
+    .. warning: implement the x and y keywords are not yet implemented.
+
+    Parameters
+    ----------
+    x : `~astropy.units.quantities` <deg>
+        The angle at which the psf is returned in the horizontal direction.
+    y : `~astropy.units.quantities` <deg>
+        The angle at which the psf is returned in the vertical direction.
+    scale : float
+        The pixel scale (e.g. arcsec / pixel). Should be set to match the map
+        with which it will be convolved.
+
+    Returns
+    -------
+    kernel : `~astropy.convolution.Gaussian2DKernel`
+        A psf kernel, normalized. Assumes 1 arcsec pixels if scale is not set.
+
+    Examples
+    --------
+    >>> p = psf(0 * u.arcmin, 0 * u.arcmin)
+    >>> p = psf(0 * u.arcmin, 0 * u.arcmin, 2 * u.arcsec)
+    """
+    # load the PSF parameters
+    path = os.path.dirname(pyfoxsi.__file__)
+    for i in np.arange(3):
+        path = os.path.dirname(path)
+    path = os.path.join(path, 'data/')
+    params = np.loadtxt(os.path.join(path, 'psf_parameters.txt'))
+
+    offaxis_angle = np.sqrt(y ** 2 + x ** 2)
+    polar_angle = np.arctan2(y, x)
+
+    poly_params = []
+    for g in params:
+        f = np.poly1d(g)
+        poly_params.append(f(offaxis_angle))
+
+    amplitude = (poly_params[0], poly_params[1], poly_params[2])
+    width = u.Quantity([poly_params[3], poly_params[4], poly_params[5]], 'arcsec')
+    width = width / scale
+    print(width)
+    # add 90 deg to the polar angle to make the rotation angle perpendicular
+    # to the polar angle
+    kernel = amplitude[0] * Gaussian2DKernel(width[0].value) + amplitude[1] * Gaussian2DKernel(width[1].value) + amplitude[2] * Gaussian2DKernel(width[2].value)
```

Comment:
Here's the source of the discrepancy between the Python and IDL PSFs.  `Gaussian2DKernel` already has a non-unity amplitude, because it is normalized, so the wider Gaussians are weaker than they should be.

There are several ways to fix this, but the easiest would be to multiply each kernel by `2*np.pi*width[i].value**2` to restore the unity amplitude.